app-modules/service-management/src/Jobs/ServiceMonitoringCheckJob.php

```diff
@@ -79,24 +80,41 @@ public function uniqueFor(): int
 
     public function handle(): void
     {
-        $response = Http::maxRedirects(15)
-            ->get($this->serviceMonitoringTarget->domain);
+        try {
+            $response = Http::maxRedirects(15)
+                ->get($this->serviceMonitoringTarget->domain);
+
+            $history = $this->serviceMonitoringTarget->histories()->create([
+                'response' => $response->status(),
+                'response_time' => $response->transferStats->getTransferTime() ?? 0,
+                'succeeded' => $response->status() === 200,
+            ]);
+
+            if ($response->status() !== 200) {
+                /** @var Collection<int, User> $recipients */
+                $recipients = $this->serviceMonitoringTarget->users()->get();
+
+                $this->serviceMonitoringTarget->teams()->each(function ($team) use ($recipients) {
+                    /** @var Collection<int, User> $users */
+                    $users = $team->users()->get();
+
+                    $recipients->concat($users)->unique();
+                });
+
+                Notification::send($recipients, new ServiceMonitoringNotification($history));
+            }
+        } catch (ConnectionException $e) {
+            $history = $this->serviceMonitoringTarget->histories()->create([
+                'response' => 523,
+                'response_time' => 0,
+                'succeeded' => false,
+            ]);
 
-        $history = $this->serviceMonitoringTarget->histories()->create([
-            'response' => $response->status(),
-            'response_time' => $response->transferStats->getTransferTime() ?? 0,
-            'succeeded' => $response->status() === 200,
-        ]);
-
-        if ($response->status() !== 200) {
-            /** @var Collection<int, User> $recipients */
             $recipients = $this->serviceMonitoringTarget->users()->get();
 
-            $this->serviceMonitoringTarget->teams()->each(function ($team) use ($recipients) {
-                /** @var Collection<int, User> $users */
+            $this->serviceMonitoringTarget->teams()->each(function ($team) use (&$recipients) {
                 $users = $team->users()->get();
-
-                $recipients->concat($users)->unique();
+                $recipients = $recipients->merge($users)->unique('id');
             });
 
             Notification::send($recipients, new ServiceMonitoringNotification($history));
```

Comment:
Because this code is pretty much the exact same as within the status check. Do you think we can turn this into a protected method in this class that can be re-used in each place?